### PR TITLE
Inject GCP Maven mirror config

### PR DESCRIPTION
**Pull request by BanzaiMan**
To mitigate the effects of https://www.traviscistatus.com/incidents/mm020jn4nzcn.

### CODE REVIEW

File: lib/travis/build/appliances/maven_central_mirror.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'travis/build/appliances/base'
+module Travis
+  module Build
+    module Appliances
+      class MavenCentralMirror < Base
+        GOOGLE_MAVEN_MIRROR='https://maven-central.storage-download.googleapis.com/repos/central/data/'
```
Comment (joshk) [inline lib/travis/build/appliances/maven_central_mirror.rb]: can this please be configurable via config as we might not want to set a mirror by default

Comment (joshk) [reply lib/travis/build/appliances/maven_central_mirror.rb]: especially for on prem customers

Comment (BanzaiMan) [reply lib/travis/build/appliances/maven_central_mirror.rb]: sure. that's reasonable. we should first test this is a feasible solution to the problem.

**Comment by BanzaiMan** [comment]
For the future reference: the information was culled from https://maven-central.storage-download.googleapis.com/index.html.

**Pull request merged**